src/annotator/components/buckets.js

```diff
@@ -0,0 +1,141 @@
+import { createElement } from 'preact';
+import classnames from 'classnames';
+import propTypes from 'prop-types';
+import scrollIntoView from 'scroll-into-view';
+
+import { setHighlightsFocused } from '../highlighter';
+import { findClosestOffscreenAnchor } from '../util/buckets';
+
+/**
+ * @typedef {import('../../types/annotator').AnnotationData} AnnotationData
+ * @typedef {import('../util/buckets').Bucket} Bucket
+ */
+
+/**
+ * A left-pointing indicator button that, when hovered or clicked, highlights
+ * or selects associated annotations.
+ *
+ * @param {Object} props
+ *  @param {Bucket} props.bucket
+ *  @param {(annotations: AnnotationData[], toggle: boolean) => any} props.onSelectAnnotations
+ *
+ */
+function BucketButton({ bucket, onSelectAnnotations }) {
+  const annotations = bucket.anchors.map(anchor => anchor.annotation);
+  const buttonTitle = `Select nearby annotations (${bucket.anchors.length})`;
+
+  function selectAnnotations(event) {
+    event.stopPropagation();
+    onSelectAnnotations(annotations, event.metaKey || event.ctrlKey);
+  }
+
+  function setFocus(focusState) {
+    bucket.anchors.forEach(anchor => {
+      setHighlightsFocused(anchor.highlights || [], focusState);
+    });
+  }
+
+  return (
+    <button
+      className="bucket-button bucket-button--left"
+      onClick={event => selectAnnotations(event)}
+      onMouseMove={() => setFocus(true)}
+      onMouseOut={() => setFocus(false)}
+      onBlur={() => setFocus(false)}
+      title={buttonTitle}
+      aria-label={buttonTitle}
```

Comment:
This button has a title and an `aria-label`. I'd very much like to give it an `aria-pressed`, but that kind of opens Pandora's Box (short version: we don't have any meaningful state on the annotator side around which annotations are currently highlighted or selected).